app/lib/timer.rb

```diff
@@ -0,0 +1,41 @@
+class Timer
+  # Util function to store and report timing information to log and metrics platform.
+  # Has support to handle multiple intermediary split times.
+
+  def initialize(prefix, tags = nil)
+    @prefix = prefix
+    @start_timestamp = Process.clock_gettime(Process::CLOCK_MONOTONIC)
+    @splits = []
+    @tags = tags || []
+  end
+
+  def add_tags(tags)
+    @tags += tags
+  end
+
+  def split(name)
+    timestamp = Process.clock_gettime(Process::CLOCK_MONOTONIC)
+    @splits << [name, timestamp]
+
+    if @splits.length > 1
+      elapsed = timestamp - (@splits.length > 1 ? @splits[-2][1] : @start_timestamp)
```

Comment:
`if @splits.length > 1` and then checking again `@splits.length > 1 ?`  (?)

Reply:
Good catch, this was just for debug since I log everything in the end.